hook!-full-game/src/singletons/CheckpointUtils.gd

```diff
@@ -0,0 +1,52 @@
+"""
+Utility functions to manage checkpoints and its file
+"""
+extends Node
+
+const FILE_PATH := "user://checkpoints.save"
```

Comment:
Saving the game wasn't part of the task.

Then, checkpoints shouldn't save themselves to a file separately from the rest of the game.

Lastly, again, you shouldn't use a singleton to grab a node and save the game.